soc/arm/st_stm32/stm32g0/power.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2021 STMicroelectronics.
+ * Copyright (c) 2021 G-Technologies Sdn. Bhd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <pm/pm.h>
+#include <soc.h>
+#include <init.h>
+
+#include <stm32g0xx_ll_utils.h>
+#include <stm32g0xx_ll_bus.h>
+#include <stm32g0xx_ll_cortex.h>
+#include <stm32g0xx_ll_pwr.h>
+#include <stm32g0xx_ll_rcc.h>
+#include <stm32g0xx_ll_system.h>
+#include <clock_control/clock_stm32_ll_common.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/* Invoke Low Power/System Off specific Tasks */
+void pm_power_state_set(struct pm_state_info info)
+{
+	if (info.state != PM_STATE_SUSPEND_TO_IDLE) {
+		LOG_DBG("Unsupported power state %u", info.state);
+		return;
+	}
+
+	switch (info.substate_id) {
+	case 1: /* this corresponds to the STOP0 mode: */
+		/* enter STOP0 mode */
+		LL_PWR_SetPowerMode(LL_PWR_MODE_STOP0);
+		LL_LPM_EnableDeepSleep();
+		/* enter SLEEP mode : WFE or WFI */
+		k_cpu_idle();
+		break;
+	case 2: /* this corresponds to the STOP1 mode: */
+		/* enter STOP1 mode */
+		LL_PWR_SetPowerMode(LL_PWR_MODE_STOP1);
+		LL_LPM_EnableDeepSleep();
+		/* enter SLEEP mode : WFE or WFI */
+		k_cpu_idle();
+		break;
+	default:
+		LOG_DBG("Unsupported power state substate-id %u",
+			info.substate_id);
+		break;
+	}
+}
+
+/* Handle SOC specific activity after Low Power Mode Exit */
+void pm_power_state_exit_post_ops(struct pm_state_info info)
+{
+	if (info.state != PM_STATE_SUSPEND_TO_IDLE) {
+		LOG_DBG("Unsupported power substate-id %u", info.state);
+	} else {
+		switch (info.substate_id) {
+		case 1:	/* STOP0 */
```

Comment:
a good reference is the stm32cube example for entering low power modes:
```
  /** Request to enter STOP 0 mode
    * Following procedure describe in STM32G0xx Reference Manual
    * See PWR part, section Low-power modes, STOP 0 mode
    */
  /* Set STOP 0 mode when CPU enters deepsleep */
  LL_PWR_SetPowerMode(LL_PWR_MODE_STOP0);
  
  /* Set SLEEPDEEP bit of Cortex System Control Register */
  LL_LPM_EnableDeepSleep();
```

Reply:
>This is a node to add for the stm32g0 serie:

This is in my other pr #36431, should I add it in this pr as well?

Reply:
>a good reference is the stm32cube example for entering low power modes:
This is already the case for entering STOP0